contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--4.2.0--4.3.0.sql

```diff
@@ -6845,6 +6899,179 @@ FROM sys.objects so
+  CAST(0 as smallint) as filestream_filegroup_id,
+  CAST(0 as sys.tinyint) as data_compression,
+  CAST('NONE' as sys.nvarchar(60)) as data_compression_desc,
+  CAST(0 as sys.bit) as xml_compression,
+  CAST('OFF' as sys.varchar(3)) as xml_compression_desc
+FROM pg_index idx
+INNER JOIN index_id_map imap on imap.indexrelid = idx.indexrelid
+INNER JOIN pg_class t on t.oid = idx.indrelid and t.relkind = 'r' and t.relispartition = false
+INNER JOIN pg_namespace nsp on t.relnamespace = nsp.oid
+INNER JOIN sys.babelfish_namespace_ext ext on (nsp.nspname = ext.nspname and ext.dbid = sys.db_id())
+where idx.indislive
+and has_schema_privilege(t.relnamespace, 'USAGE')
+
+UNION ALL
+-- entries for partitions of partitioned tables
+SELECT
+  CAST(pgi.inhrelid as sys.BIGINT) as partition_id,
+  CAST(pgi.inhparent as int) as object_id,
+  CAST(0 as int) as index_id,
+  CAST(row_number() over(partition by pgi.inhparent order by ctbl.relname) as int) as partition_number,
+  CAST(0 as sys.bigint) AS hobt_id,
+  CAST(case when ctbl.reltuples = -1 then 0 else ctbl.reltuples end as sys.bigint) AS rows,
+  CAST(0 as smallint) as filestream_filegroup_id,
+  CAST(0 as sys.tinyint) as data_compression,
+  CAST('NONE' as sys.nvarchar(60)) as data_compression_desc,
+  CAST(0 as sys.bit) as xml_compression,
+  CAST('OFF' as sys.varchar(3)) as xml_compression_desc
+FROM pg_inherits pgi
+INNER JOIN pg_class ctbl on (ctbl.oid = pgi.inhrelid and ctbl.relkind = 'r' and ctbl.relispartition)
+INNER JOIN pg_namespace nsp on ctbl.relnamespace = nsp.oid
+INNER JOIN sys.babelfish_namespace_ext ext on (nsp.nspname = ext.nspname and ext.dbid = sys.db_id())
+WHERE has_schema_privilege(ctbl.relnamespace, 'USAGE')
+AND has_table_privilege(ctbl.oid, 'SELECT,INSERT,UPDATE,DELETE,TRUNCATE,TRIGGER')
+
+UNION ALL
+-- entries for partitions of partitioned indexes
+SELECT
+  CAST(pgi.inhrelid as sys.BIGINT) as partition_id,
+  CAST(pidx.indrelid as int) as object_id,
+  CAST(imap.index_id as int) as index_id,
+  CAST(row_number() over(partition by pgi.inhparent order by ctbl.relname) as int) as partition_number,
+  CAST(0 as sys.bigint) AS hobt_id,
+  CAST(case when ctbl.reltuples = -1 then 0 else ctbl.reltuples end as sys.bigint) AS rows,
+  CAST(0 as smallint) as filestream_filegroup_id,
+  CAST(0 as sys.tinyint) as data_compression,
+  CAST('NONE' as sys.nvarchar(60)) as data_compression_desc,
+  CAST(0 as sys.bit) as xml_compression,
+  CAST('OFF' as sys.varchar(3)) as xml_compression_desc
+FROM pg_inherits pgi
+INNER JOIN pg_index cidx on cidx.indexrelid = pgi.inhrelid
+INNER JOIN pg_index pidx on pidx.indexrelid = pgi.inhparent
+INNER JOIN pg_class ctbl on (ctbl.oid = cidx.indrelid and ctbl.relkind = 'r' and ctbl.relispartition)
+INNER JOIN index_id_map imap on imap.indexrelid = cidx.indexrelid
+INNER JOIN pg_namespace nsp on ctbl.relnamespace = nsp.oid
+INNER JOIN sys.babelfish_namespace_ext ext on (nsp.nspname = ext.nspname and ext.dbid = sys.db_id())
+WHERE cidx.indislive
```

Comment:
Can we even do this in TSQL?
If not then we dont need cidx, right?

Reply:
As we spoke offline: Lets use imap only rather than cidx, since they are equivalent.

Reply:
I will evaluate the performance of sys.partitions along with sys.indexes and sys.index_column and accordingly will update the definition in separate PR if needed.

Reply:
Okay, then we can merge this provided that we followup on this quickly by raising a new PR